compiler/src/iree/compiler/Codegen/Common/GPU/GPUVectorDistribution.h

```diff
@@ -42,6 +48,31 @@ struct DistributionPattern : RewritePattern {
   std::optional<DistributionSignature> getOpSignature(Operation *op) const;
 };
 
+/// A rewriter for the pattern rewriting driver.
+struct VectorDistributionRewriter : public PatternRewriter,
+                                    public RewriterBase::Listener {
+  VectorDistributionRewriter(MLIRContext *ctx) : PatternRewriter(ctx) {
+    setListener(this);
+  }
+
+  bool hasEmptyWorklist() { return localWorklist.empty(); }
+
+  void clearWorklist() { return localWorklist.clear(); }
```

Comment:
To disambiguate from "worklist" used in the pattern application, maybe call this something that makes it clear that these are "new operations to be distributed". Maybe call this, `toBeDistributed`?

Reply:
done, lmk if this is what you had in mind :)